lib/orderbook/TradingPair.ts

```diff
@@ -267,6 +267,16 @@ class TradingPair {
     return maps.buy.get(orderId) || maps.sell.get(orderId);
   }
 
+  public addOrderHold = (orderOrOrderId: string | StampedOwnOrder, holdAmount: number) => {
+    const order = typeof orderOrOrderId === 'string' ? this.getOwnOrder(orderOrOrderId) : orderOrOrderId;
+    order.hold = order.hold ? order.hold + holdAmount : holdAmount;
```

Comment:
what about if `hold` exceeds `quantity`?
you're relying on higher-layer code to protect against that. I'd suggest to encapsulate that in what you're doing here, since it's not a generic update method, but rather specifically for `hold`.